src/Tests/Microsoft.Diagnostics.Monitoring.Tool.FunctionalTests/CollectionRuleAndApiTests.cs

```diff
@@ -0,0 +1,236 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Monitoring.TestCommon;
+using Microsoft.Diagnostics.Monitoring.TestCommon.Options;
+using Microsoft.Diagnostics.Monitoring.TestCommon.Runners;
+using Microsoft.Diagnostics.Monitoring.Tool.FunctionalTests.Fixtures;
+using Microsoft.Diagnostics.Monitoring.Tool.FunctionalTests.HttpApi;
+using Microsoft.Diagnostics.Monitoring.Tool.FunctionalTests.Runners;
+using Microsoft.Diagnostics.Monitoring.WebApi;
+using Microsoft.Diagnostics.Monitoring.WebApi.Models;
+using Microsoft.Diagnostics.Tools.Monitor;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.Diagnostics.Monitoring.Tool.FunctionalTests
+{
+    [TargetFrameworkMonikerTrait(TargetFrameworkMonikerExtensions.CurrentTargetFrameworkMoniker)]
+    [Collection(DefaultCollectionFixture.Name)]
+    public class CollectionRuleAndApiTests
+    {
+        private readonly IHttpClientFactory _httpClientFactory;
+        private readonly ITestOutputHelper _outputHelper;
+
+        public CollectionRuleAndApiTests(ITestOutputHelper outputHelper, ServiceProviderFixture serviceProviderFixture)
+        {
+            _httpClientFactory = serviceProviderFixture.ServiceProvider.GetService<IHttpClientFactory>();
+            _outputHelper = outputHelper;
+        }
+
+#if NET6_0_OR_GREATER
+        private const string DefaultRuleName = "FunctionalTestRule";
+        private readonly TimeSpan TraceDuration = Timeout.InfiniteTimeSpan;
+        private const string ClientBaseUrl = "http://localhost:10000";
+        private const string AppUrl = "http://+:10000";
+        private const string AspNetUrlsKey = "ASPNETCORE_Urls";
+
+        /// <summary>
+        /// Validates that an AspNetResponseStatus trigger will fire following an HTTP trace.
+        /// </summary>
+        [Theory(Skip = "These tests will fail until #3425 in the diagnostics repo is checked in.")]
+        //[Theory]
+        [InlineData(DiagnosticPortConnectionMode.Listen)]
+        public async Task CollectionRuleAndApi_AspNetResponseStatusTest(DiagnosticPortConnectionMode mode)
```

Comment:
nit: Remove `async` and return the `Task` from the `ScenarioRunner.SingleTarget` call.

Reply:
We discussed offline... the `Task` needs to be awaited so that the `TemporaryDirectory` instances are not disposed before the body of the test finishes.